app/src/main/java/org/p2p/wallet/history/ui/token/TokenHistoryFragment.kt

```diff
@@ -0,0 +1,178 @@
+import org.p2p.wallet.receive.token.ReceiveTokenFragment
+import org.p2p.wallet.send.ui.main.SendFragment
+import org.p2p.wallet.swap.ui.orca.OrcaSwapFragment
+import org.p2p.wallet.utils.args
+import org.p2p.wallet.utils.attachAdapter
+import org.p2p.wallet.utils.popBackStack
+import org.p2p.wallet.utils.replaceFragment
+import org.p2p.wallet.utils.showErrorDialog
+import org.p2p.wallet.utils.unsafeLazy
+import org.p2p.wallet.utils.viewbinding.viewBinding
+import org.p2p.wallet.utils.withArgs
+import kotlin.math.absoluteValue
+
+private const val EXTRA_TOKEN = "EXTRA_TOKEN"
+
+class TokenHistoryFragment :
+    BaseMvpFragment<TokenHistoryContract.View, TokenHistoryContract.Presenter>(R.layout.fragment_token_history),
+    TokenHistoryContract.View {
+
+    companion object {
+        fun create(tokenForHistory: Token.Active): TokenHistoryFragment =
+            TokenHistoryFragment()
+                .withArgs(EXTRA_TOKEN to tokenForHistory)
+    }
+
+    override val presenter: TokenHistoryContract.Presenter by inject { parametersOf(tokenForHistory) }
+
+    private val tokenForHistory: Token.Active by args(EXTRA_TOKEN)
+
+    private val historyAdapter: HistoryAdapter by unsafeLazy {
+        HistoryAdapter(
+            onTransactionClicked = presenter::onItemClicked,
+            onRetryClicked = presenter::fetchNextPage
+        )
+    }
+
+    private val binding: FragmentTokenHistoryBinding by viewBinding()
+
+    private val receiveAnalytics: ReceiveAnalytics by inject()
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        binding.setupView()
+        presenter.loadHistory()
+    }
+
+    private fun FragmentTokenHistoryBinding.setupView() {
+        toolbar.setupToolbar()
+
+        totalTextView.text = tokenForHistory.getFormattedTotal(includeSymbol = true)
+        usdTotalTextView.text = tokenForHistory.getFormattedUsdTotal()
+
+        refreshLayout.setOnRefreshListener { presenter.refresh() }
+
+        actionButtonsView.setupListeners()
+
+        historyRecyclerView.setupHistoryList()
+
+        appBarLayout.addOnOffsetChangedListener(
+            AppBarLayout.OnOffsetChangedListener { _, verticalOffset ->
+                val offset = (verticalOffset.toFloat() / appBarLayout.height).absoluteValue
+                (actionButtonsView as? OnOffsetChangedListener)?.onOffsetChanged(offset)
+            }
+        )
+    }
+
+    private fun Toolbar.setupToolbar() {
+        title = tokenForHistory.tokenName
+
+        setNavigationOnClickListener { popBackStack() }
+        if (BuildConfig.DEBUG) {
+            inflateMenu(R.menu.menu_history)
+            setOnMenuItemClickListener {
+                var isHandled = false
+                if (it.itemId == R.id.closeItem) {
+                    presenter.closeAccount()
+                    isHandled = true
+                }
+                isHandled
```

Comment:
wtf? 
just return boolean directly

Reply:
well that's my code. I couldn't understand when and what should this lambda return. What does it mean to return false or return true. So I created this val to make it explicit. Can be inlined, if there is need.

Reply:
But isn't it understandable when you return it directly?
```
{
      if (someBoolean) {
            // do smth
            return@someListener true
      }
      
      return@someListener false
}
```

the boolean variable is redundant here